### PR TITLE
Homework 12

### PR DESCRIPTION
**Pull request by asish918**
# Homework 2 Asish

### Topics Covered:
- JPA Hibernate

### Drift Video Explanation
[Drift](https://video.drift.com/v/abJgdXUh2USysehMNS1QGn2KhvbUgedZud21vlXhUG90)

### Output Screenshots - 

Screenshot 2024-02-01 at 19 58 16

Screenshot 2024-02-01 at 19 58 14

Screenshot 2024-02-01 at 19 58 12

Screenshot 2024-02-01 at 19 58 10

Screenshot 2024-02-01 at 19 58 08

Screenshot 2024-02-01 at 19 58 06

Screenshot 2024-02-01 at 19 58 03

Screenshot 2024-02-01 at 19 57 54

### Sonar Report from my localhost (since github actions yml file is not working) - 

Screenshot 2024-02-01 at 20 13 28

### CODE REVIEW

File: backend/homework/12/src/main/java/asish/kdu/spring_jpa_exercise/controller/ShiftController.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    private ShiftService shiftService;
+    @Autowired
+    public ShiftController(ShiftService shiftService) {
+        this.shiftService = shiftService;
+    }
+
+    @PostMapping("/shift")
+    public ResponseEntity<String> createShift(@RequestBody Shift shift) {
+        shiftService.addShift(shift);
+        return new ResponseEntity<>("Shift created successfully", HttpStatus.CREATED);
```
Comment (rishav-kd) [inline backend/homework/12/src/main/java/asish/kdu/spring_jpa_exercise/controller/ShiftController.java:23]: lets move this to service layers

File: backend/homework/12/src/main/java/asish/kdu/spring_jpa_exercise/entities/ShiftUsers.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+@Entity
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Data
+@Table(name = "shift_user")
+public class ShiftUsers {
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    @Column(columnDefinition = "CHAR(36) DEFAULT (UUID())")
+    private String id;
```
Comment (rishav-kd) [inline backend/homework/12/src/main/java/asish/kdu/spring_jpa_exercise/entities/ShiftUsers.java:20]: Lets use UUID here

File: backend/homework/12/src/main/java/asish/kdu/spring_jpa_exercise/services/ShiftTypeService.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+public class ShiftTypeService {
+    private ShiftTypeRepository shiftTypeRepository;
+
+    @Autowired
+    public ShiftTypeService(ShiftTypeRepository shiftTypeRepository) {
+        this.shiftTypeRepository = shiftTypeRepository;
+    }
+
+    public void addShiftType(ShiftType shiftType) {
+        shiftTypeRepository.save(shiftType);
```
Comment (rishav-kd) [inline backend/homework/12/src/main/java/asish/kdu/spring_jpa_exercise/services/ShiftTypeService.java:18]: Lets add validation here.

**Review by rishav-kd — COMMENTED** [review]
Looks good. Focus on comments

**Comment by aakash232** [comment]
@asish918 
Create a PR to your forked repo's MAIN branch.

✅ `MAIN (your_name/kdu-coursework)  <- homework-1 `

❌ `MAIN (kickdrum/kdu-coursework)  <- homework-1 `

**Pull request closed without merging**